extension/test/data_test.ts

```diff
@@ -85,6 +85,67 @@ describe('data.ts', function () {
     });
   });
 
+  describe('parseWordDictEntry', function () {
+    describe('with a valid word dict line', function () {
+      it('parses a word dict entry from a word dict line which has a word, pronunciation, and definitions', function () {
```

Comment:
In these test names, we're saying that the correct outcome is that the method is going to "parse a word dict entry" but it's that leaves some ambiguity on what correct parsing is.

In this case, the key to parsing is mapping the word to the `word` property and so forth. It's a bit verbose to do it all at once so I actually recommend testing each output property in it's own test like: 'parses the word part of the entry and returns in word property'. (then you would have three similar tests with different assertions basically)

This has several benefits:
- Failures are more precise since the exact piece of extraction is called out.
- Writer intention is extremely clear so as documentation the test is improved.
- Future additions to entries can be tested primarily in new tests; it's an anti-pattern to update pre-existing tests for new functionality since that reduces their value as regression finders.

The down side is additional tests but we're not close to any size constraints on a project of this size! Writing them is very easy since the structure is the same after the first one.

Let me know what you think!

Reply:
I tried this out and it seems reasonable. The tests are more specific now and it's easier to see what specifically is failing when a test fails. Thanks!